src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -1314,8 +1337,1089 @@ public boolean mapSwitchControlValue(String value) {
+
+		// Populate reject unencrypted caller switch control
+		if (streamInfo.getPassphraseSet()) {
+			addAdvanceControlProperties(advancedControllableProperties, createSwitch(stats, streamGroup + StreamControllingMetric.REJECT_UNENCRYPTED_CALLERS.getName(), streamInfo.getStrictMode(),
+					DecoderConstant.DISABLE, DecoderConstant.ENABLE));
+		}
+	}
+
+	/**
+	 * This method is used for populate all stream control properties :
+	 * Protocol: TS over SRT
+	 * Stream conversion
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 * @param streamInfo stream config info
+	 */
+	private void populateStreamControlCaseTSOverSRTStreamConversion(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, StreamInfo streamInfo,
+			String streamGroup) {
+		boolean srtToUDP = streamInfo.getSrtToUdp();
+
+		// Populate srt to udp switch control
+		addAdvanceControlProperties(advancedControllableProperties, createSwitch(stats, streamGroup + StreamControllingMetric.SRT_TO_UDP_STREAM_CONVERSION.getName(), srtToUDP,
+				DecoderConstant.DISABLE, DecoderConstant.ENABLE));
+
+		// Populate relevant control when srt to udp is enabled
+		if (srtToUDP) {
+			addAdvanceControlProperties(advancedControllableProperties,
+					createText(stats, streamGroup + StreamControllingMetric.SRT_TO_UDP_ADDRESS.getName(), streamInfo.getSrtToUdpAddress()));
+			addAdvanceControlProperties(advancedControllableProperties,
+					createNumeric(stats, streamGroup + StreamControllingMetric.SRT_TO_UDP_PORT.getName(), checkForNullData(streamInfo.getSrtToUdpPort().toString())));
+			addAdvanceControlProperties(advancedControllableProperties,
+					createText(stats, streamGroup + StreamControllingMetric.SRT_TO_UDP_TOS.getName(), streamInfo.getSrtToUdpTos()));
+			addAdvanceControlProperties(advancedControllableProperties,
+					createNumeric(stats, streamGroup + StreamControllingMetric.SRT_TO_UDP_TTL.getName(), checkForNullData(streamInfo.getSrtToUdpTtl().toString())));
+		}
+	}
+
+	/**
+	 * This method is used for populate all stream control properties :
+	 * Protocol: TS over SRT
+	 * Encrypted
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 * @param streamInfo stream config info
+	 */
+	private void populateStreamControlCaseTSOverSRTEncrypted(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, StreamInfo streamInfo, String streamGroup) {
+		boolean encrypted = streamInfo.getPassphraseSet();
+
+		// Populate encrypted switch control
+		addAdvanceControlProperties(advancedControllableProperties, createSwitch(stats, streamGroup + StreamControllingMetric.ENCRYPTED.getName(), encrypted,
```

Comment:
did you mean addAdvancedControlProperties as a method name?

Reply:
Yes! it's the method name.